### PR TITLE
blender-hip: add blender with hip support

### PR DESCRIPTION
**Pull request by Flakebi**
###### Description of changes
As a follow-up to the rocm 5.2 update in #187005, add blender with hip support, which allows rendering on the GPU.

The hip renderer can be set in Edit > Preferences, set System > Cycles Render Devices to `HIP`.
For rendering the example cube, go to the render settings (the camera icon in the lower, right panel), set `Render Engine` to `Cycles` and `Device` to `GPU Compute`.
Then render the image (`F12`). It takes a minute or more to compile the render kernel. The compilation command gets printed to stdout by blender.

Tested on navi10 (AMD Radeon RX 5700 XT) and navi21 (RX 6700 XT).

Builds are on the flakebi cachix, so you can use

```
cachix use flakebi
nix-build -A blender-hip
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @acowley, @lovesegfault, @InternetUnexplorer

### CODE REVIEW

**Comment by veprbl** [comment]
Could you explain a bit, is there a reason why we can't enable hip by default? I see it builds on ofborg, so it's not unfree, right?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -26234,6 +26234,13 @@ with pkgs;
 
   blender-with-packages = callPackage ../applications/misc/blender/wrapper.nix {};
 
+  blender-hip = callPackage  ../applications/misc/blender {
+    # LLVM 11 crashes when compiling GHOST_SystemCocoa.mm
+    stdenv = if stdenv.isDarwin then llvmPackages_10.stdenv else stdenv;
+    inherit (darwin.apple_sdk.frameworks) Cocoa CoreGraphics ForceFeedback OpenAL OpenGL;
+    hipSupport = true;
+  };
```
Comment (veprbl) [inline pkgs/top-level/all-packages.nix:26242]:
```suggestion
  blender-hip = blender.override { hipSupport = true; }
```

**Comment by Flakebi** [comment]
Thanks for the review!

> Could you explain a bit, is there a reason why we can't enable hip by default? I see it builds on ofborg, so it's not unfree, right?

Right, everything in hip is open source (MIT I think, except llvm, which has its own license).
Enabling hip by default would be nice, it would increase closure size significantly though (because it pulls in all of hip).
`nix path-info -Sh .#blender`: 1.4 GB
`nix path-info -Sh .#blender-hip`: 4.1 GB

What do you think? Should we still enable hip support by default?

**Comment by lovesegfault** [comment]
Personally, I think the closure increase is too much and having `blender-hip` is an acceptable compromise.

**Comment by totoroot** [comment]
Thank you so much for your work on this @Flakebi :heart: 
I just got an RDNA2 GPU, namely an RX 6600XT, and was very much looking forward to using it with Blender Cycles.

I tested your changes and they worked without any issues. Both my CPU and my GPU now show up in the HIP tab and render times were cut down significantly.

The [monk](https://download.blender.org/demo/cycles/lone-monk_cycles_and_exposure-node_demo.blend) demo from Blender rendered in just over ten minutes on default settings using the RX 6600XT with HIP, compared to close to about 1h30m with my hexa-core Ryzen 3600X.

 

Thanks again for contributing to making NixOS once again the perfect power-user Linux distribution (at least for me) :clap:

**Pull request merged**